### ISSUE TITLE
Animated Textures are broken in Tinkers' Construct Smeltery

### ISSUE THREAD

**Issue by Desempregago** [LoliKingdom/LoliASM#63]
Plus, I don't know what I can toggle in the config to alleviate this

https://user-images.githubusercontent.com/14002834/148842538-4807f8d2-b856-4aa1-808d-569536f2b439.mp4

**Comment by Desempregago** [comment]
The molten metal is supposed to be flowing down, not being static

**Comment by Desempregago** [comment]
Using LoliASM v4.4

**Comment by Rongmario** [comment]
Try disabling `releaseSpriteFramesCache`.

**Comment by xJon** [comment]
Can confirm this is happening also with version 4.8.

**Comment by Rongmario** [comment]
Even when `releaseSpriteFramesCache` is false?

**Comment by xJon** [comment]
@Rongmario Sorry for the misunderstanding - I did not test it beyond the default configuration, I will try to with your suggestion soon.

**Comment by Rongmario** [comment]
That's no problem.

**Comment by xJon** [comment]
@Rongmario I can confirm this is still an issue with `releaseSpriteFramesCache` disabled.

**Comment by Rongmario** [comment]
What about disabling `squashBakedQuads`?

**Comment by xJon** [comment]
@Rongmario Still an issue with `squashBakedQuads` disabled.

**Comment by Rongmario** [comment]
Quite marvellous. I hate Tinkers' Construct.

**Comment by xJon** [comment]
@Rongmario Wait, I think more testing is necessary - I just completely removed LoliASM and this is still an issue 👀

**Comment by Rongmario** [comment]
Do you have VanillaFix installed?

**Comment by xJon** [comment]
I do - and I can see where this is going

**Comment by Rongmario** [comment]
Do you have `textureFixes` enabled for VF? Provided you're not using the dynamic model loading version of VF.

**Comment by xJon** [comment]
I do have `textureFixes` enabled, and I can see that this issue with VanillaFix and Tinkers animated textures was reported in the past but I guess was never fully resolved - https://github.com/DimensionalDevelopment/VanillaFix/issues/39#issuecomment-439508499

**Comment by Rongmario** [comment]
Yeah, I'm going to fix this soon, got way too many projects going on atm. Even a VanillaFix fork/rework is on the table.

**Comment by xJon** [comment]
@Rongmario No worries, this is not your or your mod's responsibility, but thank you for the will to help make modded Minecraft better! If you do rework VanillaFix I'd be sure to include it in my modpacks, and I am currently testing LoliASM.
About this issue, you can probably close it as it is a VanillaFix issue.

**Comment by Rongmario** [comment]
Thank you, but I think I'll keep it open for the time being as I think `releaseSpriteFramesCache` may have the same effect as VF's `textureFixes`. However, they don't quite do the same thing.

Also, I'm joined with quite a few others on trying to make 1.12.2 modded better w/ more support, we are quite ambitious: https://cleanroommc.com/

**Comment by xJon** [comment]
I tested Tinkers' Construct with LoliASM, without VanillaFix, and this issue is not present - the textures animate properly.
And, that's awesome - I joined your Discord server!

**Comment by Rongmario** [comment]
Oh? Good news. I always thought LoliASM had something to do with it...

**Comment by embeddedt** [comment]
For the record, I believe disabling `textureFixes` from VanillaFix will reduce the FPS substantially on AMD GPUs due to the fact that reuploading the textures each frame is extremely slow on them, unless `releaseSpriteFramesCache` does the same thing. Even on my Nvidia card I have found this VanillaFix optimization to actually be quite effective; when profiling the slowness in 1.16 I noticed that uploading animated textures took up a substantial part of the frame time. The VanillaFix beta for 1.16 made that issue go away.
